data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/PeerClientPool.java

```diff
@@ -32,8 +46,16 @@ public void setClientTimeoutSeconds(int clientTimeoutSeconds) {
 
     private TraceServiceGrpc.TraceServiceBlockingStub createGRPCClient(final String ipAddress) {
         // TODO: replace hardcoded port with customization
-        return Clients.builder(String.format("gproto+http://%s:21890/", ipAddress))
-                .writeTimeout(Duration.ofSeconds(clientTimeoutSeconds))
-                .build(TraceServiceGrpc.TraceServiceBlockingStub.class);
+        final ClientBuilder clientBuilder;
+        if (ssl) {
+            clientBuilder = Clients.builder(String.format("%s://%s:21890/", GRPC_HTTPS, ipAddress))
```

Comment:
Nit: I have seen people advocate for using StringBuilder in these situations. Not a blocker at all.

Reply:
Looks like StringBuilder is going to take more lines. Will revisit when the port is also customized.